internal/handlers/common/distinct.go

```diff
@@ -30,68 +29,33 @@ import (
 //
 //nolint:vet // for readability
 type DistinctParams struct {
-	DB         string
-	Collection string
-	Key        string
-	Filter     *types.Document
-	Comment    string
+	DB         string          `ferretdb:"$db"`
+	Collection string          `ferretdb:"collection"`
+	Key        string          `ferretdb:"key"`
+	Filter     *types.Document `ferretdb:"query,opt"`
+	Comment    string          `ferretdb:"comment,opt"`
+
+	Collation *types.Document `ferretdb:"collation,unimplemented"`
+
+	ReadConcern *types.Document `ferretdb:"readConcern,ignored"`
 }
 
 // GetDistinctParams returns `distinct` command parameters.
 func GetDistinctParams(document *types.Document, l *zap.Logger) (*DistinctParams, error) {
-	var err error
-
-	unimplementedFields := []string{
-		"collation",
-	}
-	if err = Unimplemented(document, unimplementedFields...); err != nil {
-		return nil, err
-	}
-
-	ignoredFields := []string{
-		"readConcern",
-	}
-	Ignored(document, l, ignoredFields...)
-
 	var dp DistinctParams
 
-	if dp.DB, err = GetRequiredParam[string](document, "$db"); err != nil {
-		return nil, err
-	}
-
-	collectionParam, err := document.Get(document.Command())
+	err := commonparams.ExtractParams(document, "distinct", &dp, l)
```

Comment:
I think you want to catch `ErrFieldNotPopulated` for missing `key`

Reply:
Make it handled inside checkAllRequiredFIeldsPopulated.